lib/etcdv3.rb

```diff
@@ -45,6 +45,23 @@ def token
     request.token
   end
 
+  def key
+    File.read(File.expand_path(@options[:key])) if options.has_key?(:key)
+  end
+
+  def cert
+    File.read(File.expand_path(@options[:cert])) if options.has_key?(:cert)
+  end
+
+  def cacert
+    File.read(File.expand_path(@options[:cacert])) if options.has_key?(:cacert)
+  end
+
+  def tls_creds
+    # The order of the elements in the array is important
+    [cacert, key, cert].compact
+  end
+
```

Comment:
This seems to work pretty well, my only nit is the loose nature of the tls_creds method.  If I am doing just server authentication, I could pass my `cacert` in as `cert, or `key` and it would still work.  There may be value in being a bit more strict here.  

Also what are your thoughts on throwing an exception when `key` is present without `cert` and vice versa?